qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -412,16 +457,26 @@ def x_reader(conn, callback):
 parser.add_argument('--notify-monitor-layout', action='store_true',
                     help='Notify running instance in --watch mode'
                          ' about changed monitor layout')
+# Add it for the help only
+parser.add_argument('--force', action='store_true', default=False,
+                    help='Force running daemon without enabled services'
+                         ' \'guivm-gui-agent\' or \'audiovm-audio-agent\'')
 
 
 def main(args=None):
-    """ Main function of qvm-start-gui tool"""
+    """ Main function of qvm-start-daemon tool"""
+    only_if_service_enabled = ['guivm-gui-agent', 'audiovm-audio-agent']
+    enabled_services = [service for service in only_if_service_enabled if
+                        os.path.exists('/var/run/qubes-service/%s' % service)]
```

Comment:
@marmarek: that does not work under dom0. Maybe looking for services in 'features' would be better?

Reply:
Besides the need for https://github.com/QubesOS/qubes-core-admin/pull/322, what would enable that service on dom0 in the classic case of guivm=dom0? In case of separate guivm, salt would sent required services, but not in case of dom0 (including in-place upgrade).
Maybe this tool should start in dom0 regardless of services enabled? In case of "true guivm" (with pci passthrough) there will be no X session in dom0, so relevant .desktop files won't be started at all and it will avoid running this part (useless then) in dom0.

Also, in case of non-dom0, this is not fully accurate - with other types of guivm (the one with VNC, and also the one with GPU passthrough) qvm-start-daemon should also start - not only for guivm on top of gui-agent. This may mean a need for a separate service.

Reply:
Yes indeed. Nothing will set it up to post anaconda commands. So better run it unconditionally for dom0 only at the moment.